lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,29 +1,921 @@
+                l1_ratio=l1_ratio, normalize=True)
+
+        alpha = alpha_init
+        alpha_step_factor = 1.1 # being by increasing alpha by 10%
+        for idx in np.arange(max_iter):
+
+            metric = self._apply_fit(X, y, alpha)
+
+            # Check goodness metric
+            if (metric > target_score):
+                # Over-fitting occured, increase alpha
+                alpha *= alpha_step_factor
+            else:
+                # We're good, exit loop
+                break
+        
+        self.introduced_noise_score = metric
+
+
+        return self.corrected_lc
+
+
+
+
+    def over_fitting_metric(self):
+        """ Uses a LombScarglePeriodogram to assess the change in broad-band
+        power in a corrected light curve to measure degree of over-fitting
+
+        This function expects a median normalized light curve
+        """
+
+        # TODO: handle gaps and/or masks
+
+        pgOrig = self.lc.to_periodogram(freq_unit=u.microHertz, maximum_frequency=400, minimum_frequency=10)
+        pgCorrected = self.corrected_lc.to_periodogram(freq_unit=u.microHertz, maximum_frequency=400, minimum_frequency=10)
+
+        # Get an esitmate of the PSD at the uncertainties limit
+        # The raw and corrected uncertainties should be essentially identical so use the corrected
+        # TODO: the periodogram of WGN should be analytical to compute!
+        nNonGappedCadences = len(self.lc.flux)
+        meanCorrectedUncertainties = np.mean(self.corrected_lc.flux_err)
+        WGNCorrectedUncert = (np.random.randn(nNonGappedCadences,1) * meanCorrectedUncertainties).T[0]
+        model_err   = np.zeros(len(self.lc.flux))
+        noise_lc = LightCurve(self.lc.time, WGNCorrectedUncert, model_err)
+        pgCorrectedUncert = noise_lc.to_periodogram(freq_unit=u.microHertz, maximum_frequency=400, minimum_frequency=10)
+        meanCorrectedUncertPower = np.mean(np.array(pgCorrectedUncert.power))
+        
+        # Compute the change in power
+        pgChange = np.array(pgCorrected.power) - np.array(pgOrig.power)
+
+        # If no increase in power in ANY bands then return a perfect loss
+        # function
+        if (len(np.nonzero(pgChange>0.0)[0]) == 0):
+            return 0.0
+        
+        # We are only concerned with bands where the power increased so
+        # when(pgCorrected - pgRaw) > 0 
+        # Normalize by the noise in the uncertainty
+        # We want the goodness to begin to degrade when the introduced noise is greater than the uncertainties.
+        # So, when DeltaNoise > 0.5 (given twiceSigmoidInv defn.)
+        metric = np.sum(pgChange[pgChange>0.0]) / ((len(np.nonzero(pgChange>0.0)[0]))*meanCorrectedUncertPower)
+
+        
+      # #****************
+      # # This converts the loss function into a goodness metric
+      # # Adjust the decay rate of the sigmoid so that it's not too steep of a decent.
+      # # A noiseScale of about 0.05 looks good.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [indentation is not a multiple of four (comment)](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4896575)